### PR TITLE
[bugs] zstdgrep/grep inconsistencies

### PR DESCRIPTION
**Pull request by bimbashrestha**
Fixing zstdgrep inconsistencies listed on https://github.com/facebook/zstd/issues/2064

See each commit message for details

* Fix multiple -e
* Fix pipe in with -f
* Fix "incorrect parameters" output on regexp
* Don't show filename on single file (ie: no to /tmp/file:start and yes to start <- grep does this)
* Fixing --regexp giving lots of irelevant complaints

### CODE REVIEW

**Comment by Cyan4973** [comment]
Could we add some test case for this new capability ?

**Comment by bimbashrestha** [comment]
FreeBSD failed saying it can't find zstdcat? I'm going to re-run it just to make sure it's not just being strange

**Comment by bimbashrestha** [comment]
Nope, that's a real error:(

File: tests/playTests.sh [hunk 1]
```diff
@@ -254,6 +254,14 @@ ZCAT=./zstdcat $ZSTDGREP 2>&1 "1234" tmp_grep_bad.zst && die "Should have failed
 ZCAT=./zstdcat $ZSTDGREP 2>&1 "1234" tmp_grep_bad.zst | grep "No such file or directory" || true
 rm -f tmp_grep*
 
+println "\n===> zstdgrep mulitple -e"
```
Comment (Cyan4973) [inline tests/playTests.sh:257]: nit : `mulitple`

**Comment by Cyan4973** [comment]
One may need to setup a local FreeBSD VM to reproduce the issue and better diagnose it.

**Comment by bimbashrestha** [comment]
> One may need to setup a local FreeBSD VM to reproduce the issue and better diagnose it.

I'll try that quick guess fix first. If that doesn't work, I'll setup of a local VM:/

**Comment by bimbashrestha** [comment]
Okay that seemed to fix it. It was just a matter of adding ZCAT= in front. I'm going to include some other fixes from that issue in this PR too

**Comment by bimbashrestha** [comment]
I tested zstdgrep on the list from the issue and these changes take care of everything but the bash stuff at the end. These: 

```
# Process substitutions with bash not working as with grep on its own

$ cp  /tmp/file /tmp/file2
$ grep --file=<( echo start ) /tmp/file /tmp/file2
/tmp/file:start
/tmp/file2:start
$ grep -f <( echo start ) /tmp/file /tmp/file2
/tmp/file:start
/tmp/file2:start
$ zstdgrep --file=<( echo start ) /tmp/file /tmp/file2
/tmp/file:start
/tmp/file2:start
$ zstdgrep -f <( echo start ) /tmp/file /tmp/file2
/tmp/file:start
```

**Comment by bimbashrestha** [comment]
Okay, I think this is ready to merge. The only unsupported functionality are the the bash commands listed at the end of that issue but it looks like zgrep produces strange results for those too (so I don't think its worth trying to fix them for zstdgrep).

**Comment by Cyan4973** [comment]
Could we add a few tests showing off the new capabilities ?

**Comment by bimbashrestha** [comment]
Ah it looks like I deleted the changes to playTests when I force pushed. Yes. I"ll add those back

**Comment by Cyan4973** [comment]
Looks good to me !

**Pull request merged**